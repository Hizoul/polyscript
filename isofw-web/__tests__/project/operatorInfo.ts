import SharedOperatorInfo from "isofw-shared/src/components/project/operatorInfo"
import operatorInfoTest from "isofw-shared/src/tests/project/operatorInfo"
import CurrentOperatorDisplay from "isofw-web/src/components/project/operatorInfo/currentOperatorDisplay"

operatorInfoTest(SharedOperatorInfo(CurrentOperatorDisplay))
