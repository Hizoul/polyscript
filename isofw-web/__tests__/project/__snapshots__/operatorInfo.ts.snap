// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` operator info test  works as planned:  creation of cameras  1`] = `
Array [
  Object {
    "_id": "c1bbbbbbbbbbbbbbbbbbbbbb",
    "ip": "localhost:4202/c1",
    "name": "Camera#1",
  },
  Object {
    "_id": "c2bbbbbbbbbbbbbbbbbbbbbb",
    "ip": "localhost:4202/c2",
    "name": "Camera#2",
  },
  Object {
    "_id": "c3bbbbbbbbbbbbbbbbbbbbbb",
    "ip": "localhost:4202/c3",
    "name": "Camera#3",
  },
]
`;

exports[` operator info test  works as planned:  creation of proInjects  1`] = `
Array [
  Object {
    "_id": "b19999999999999999999999",
    "cameras": Array [
      "c1bbbbbbbbbbbbbbbbbbbbbb",
      "c2bbbbbbbbbbbbbbbbbbbbbb",
      "c3bbbbbbbbbbbbbbbbbbbbbb",
    ],
    "name": "pn1",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
          "c3bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u1",
      },
      Object {
        "cameras": Array [
          "c2bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u2",
      },
    ],
    "operators": Array [
      "u1",
      "u2",
    ],
    "shot": 1,
  },
  Object {
    "_id": "b29999999999999999999999",
    "cameras": Array [
      "c1bbbbbbbbbbbbbbbbbbbbbb",
      "c2bbbbbbbbbbbbbbbbbbbbbb",
    ],
    "name": "pn2",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u3",
      },
      Object {
        "cameras": Array [
          "c2bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u1",
      },
    ],
    "operators": Array [
      "u1",
      "u3",
    ],
    "shot": 8,
  },
  Object {
    "_id": "b39999999999999999999999",
    "cameras": Array [
      "c1bbbbbbbbbbbbbbbbbbbbbb",
      "c3bbbbbbbbbbbbbbbbbbbbbb",
    ],
    "name": "pn3",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u3",
      },
      Object {
        "cameras": Array [
          "c3bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "operator": "u2",
      },
    ],
    "operators": Array [
      "u2",
      "u3",
    ],
    "shot": 6,
  },
]
`;

exports[` operator info test  works as planned: Before anything 1`] = `
<div>
  <div
    className="Link"
  >
    {
  "popoverOpen": ".operatorChooser",
  "className": "subtitleBox centerText"
}
    <div
      className="subtitle"
    >
      none selected
    </div>
    <span>
       
    </span>
  </div>
  <div
    className="Popover"
  >
    {
  "className": "operatorChooser"
}
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {
  "popoverClose": ".operatorChooser"
}
        <div
          slot="title"
        >
          No one
        </div>
        <div
          slot="footer"
        >
          All cameras
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
            <span />
          </span>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` operator info test  works as planned: Invalid user selected 1`] = `
<div>
  <div
    className="Link"
  >
    {
  "popoverOpen": ".operatorChooser",
  "className": "subtitleBox centerText"
}
    <div
      className="subtitle"
    >
      <span />
    </div>
    <span>
       
    </span>
  </div>
  <div
    className="Popover"
  >
    {
  "className": "operatorChooser"
}
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {
  "popoverClose": ".operatorChooser"
}
        <div
          slot="title"
        >
          No one
        </div>
        <div
          slot="footer"
        >
          All cameras
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
            <span />
          </span>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` operator info test  works as planned: Removed user selection 1`] = `
<div>
  <div
    className="Link"
  >
    {
  "popoverOpen": ".operatorChooser",
  "className": "subtitleBox centerText"
}
    <div
      className="subtitle"
    >
      none selected
    </div>
    <span>
       
    </span>
  </div>
  <div
    className="Popover"
  >
    {
  "className": "operatorChooser"
}
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {
  "popoverClose": ".operatorChooser"
}
        <div
          slot="title"
        >
          No one
        </div>
        <div
          slot="footer"
        >
          All cameras
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
            <span />
          </span>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[` operator info test  works as planned: first user selected 1`] = `
<div>
  <div
    className="Link"
  >
    {
  "popoverOpen": ".operatorChooser",
  "className": "subtitleBox centerText"
}
    <div
      className="subtitle"
    >
      <span />
    </div>
    <span>
      CA 
      <span />
      <span />
    </span>
  </div>
  <div
    className="Popover"
  >
    {
  "className": "operatorChooser"
}
    <div
      className="List"
    >
      {}
      <div
        className="ListItem"
      >
        {
  "popoverClose": ".operatorChooser"
}
        <div
          slot="title"
        >
          No one
        </div>
        <div
          slot="footer"
        >
          All cameras
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
            <span />
          </span>
        </div>
      </div>
      <div
        className="ListItem"
      >
        {
  "popoverClose": true
}
        <div
          slot="title"
        >
          <span />
        </div>
        <div
          slot="footer"
        >
          <span>
            Cameras: 
            <span />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
